backend/packages/Upgrade/src/api/controllers/validators/StratificationValidator.ts

```diff
@@ -0,0 +1,24 @@
+import { IsNotEmpty, IsObject, IsOptional, IsString } from 'class-validator';
+
+export class FactorStrata {
+  @IsNotEmpty()
+  @IsString()
+  public factor: string;
+
+  @IsObject()
+  public value: Record<string, number>;
+}
+
+export class StratificationInputValidator {
+  @IsNotEmpty()
+  @IsString()
+  public userId: string;
+
+  @IsNotEmpty()
+  @IsString()
+  public factor: string;
+
+  @IsOptional()
+  @IsString()
+  public value?: string;
```

Comment:
above we say value can be a number, and it is required, here we have string, and it's optional. I imagine these are closely related. It doesn't seem like this can be optional though, can someone explain these new models with an example or two?

Reply:
This value is renamed `stratificationFactorValue` which gets stored in DB.